partiql-ast/partiql-ast-macros/src/lib.rs

```diff
@@ -0,0 +1,136 @@
+use darling::{FromDeriveInput, FromField, FromVariant};
+use inflector::Inflector;
+use proc_macro2::TokenStream;
+use quote::{format_ident, quote};
+use syn::{Data, Fields};
+use syn::{DeriveInput, Ident};
+
+#[proc_macro_derive(Visit, attributes(visit))]
+pub fn visit_derive(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    let ast = syn::parse_macro_input!(input as syn::DeriveInput);
+    let gen = impl_visit(&ast);
+    gen.into()
+}
+
+#[derive(FromDeriveInput)]
+#[darling(attributes(visit))]
+struct VisitItemOptions {
+    #[darling(rename = "skip_recurse")]
+    should_skip_recurse: Option<bool>,
+}
+
+fn should_skip_recurse(input: &syn::DeriveInput) -> bool {
+    VisitItemOptions::from_derive_input(input)
+        .expect("parse meta")
+        .should_skip_recurse
+        .unwrap_or(false)
+}
+
+#[derive(FromField, FromVariant)]
+#[darling(attributes(visit))]
+struct VisitFieldOptions {
+    #[darling(rename = "skip")]
+    should_skip: Option<bool>,
+}
+
+fn should_skip_field(field: &syn::Field) -> bool {
+    VisitFieldOptions::from_field(field)
+        .expect("parse meta")
+        .should_skip
+        .unwrap_or(false)
+}
+
+fn should_skip_variant(variant: &syn::Variant) -> bool {
+    VisitFieldOptions::from_variant(variant)
+        .expect("parse meta")
+        .should_skip
+        .unwrap_or(false)
+}
+
+fn impl_visit(ast: &syn::DeriveInput) -> proc_macro2::TokenStream {
+    let visit_fn_name = &ast.ident.to_string().to_snake_case();
+    let enter_fn_name = Ident::new(
+        &format!("enter_{}", visit_fn_name),
+        proc_macro2::Span::call_site(),
+    );
+    let exit_fn_name = Ident::new(
+        &format!("exit_{}", visit_fn_name),
+        proc_macro2::Span::call_site(),
+    );
+
+    let visit_children = (!should_skip_recurse(ast)).then(|| impl_visit_children(&ast));
+
+    let ast_name = &ast.ident;
+    quote! {
```

Comment:
I'm still new to procedure macros in Rust. Is there a way to see what the procedural macro generates in `cargo` or CLion?

Reply:
`cargo expand` in the directory of the crate using the macro will expand *all* macro usages.